src/blueapi/service/model.py

```diff
@@ -169,6 +170,37 @@ class EnvironmentResponse(BlueapiBaseModel):
     )
 
 
+class SourceInfo(str, Enum):
+    pypi = "pypi"
+    scratch = "scratch"
+
+    def __str__(self):
+        return self.value
+
+
+class PackageInfo(BlueapiBaseModel):
+    name: str = Field(description="Name of the package", default_factory=str)
+    version: str = Field(description="Version of the package", default_factory=str)
+    location: str = Field(description="Location of the package", default_factory=str)
+    is_dirty: bool = Field(
+        description="Does the package have uncommitted changes", default_factory=bool
+    )
+    source: SourceInfo = Field(
+        description="Source of the package", default=SourceInfo.pypi
+    )
+
+
+class ScratchResponse(BlueapiBaseModel):
+    """
+    State of the scratch area.
+    """
+
+    installed_packages: list[PackageInfo] = Field(
+        description="List of installed packages", default_factory=list
+    )
+    enabled: bool = Field(description="Scratch area state", default=False)
```

Comment:
Should: Since it's no longer just about the scratch area I think there's some renaming needed here:

```suggestion
class PythonEnvironmentResponse(BlueapiBaseModel):
    """
    State of the scratch area.
    """

    installed_packages: list[PackageInfo] = Field(
        description="List of installed packages", default_factory=list
    )
    scratch_enabled: bool = Field(description="Scratch area state", default=False)
```

Should also rename the REST endpoint